lib/Doctrine/ODM/MongoDB/ChangeSet/CollectionChangeSet.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\ODM\MongoDB\ChangeSet;
+
+use Doctrine\ODM\MongoDB\PersistentCollection\PersistentCollectionInterface;
+
+class CollectionChangeSet
+{
+    /**
+     * @var PersistentCollectionInterface
+     */
+    private $collection;
+
+    /**
+     * @var array
+     */
+    private $changes;
+
+    /**
+     * @param PersistentCollectionInterface $collection
+     * @param ObjectChangeSet[] $changes
+     */
+    public function __construct(PersistentCollectionInterface $collection, array $changes)
```

Comment:
One thing we'll have to think of is the ability to replace collections:
```
$object->embedMany = []
```

This won't just be a collection changeset but instead the entire collection gets replaced with a new collection. In order to properly populate insertedDocuments and deletedDocuments we'd have to know about both collections.